test/test_pytree.py

```diff
@@ -328,9 +328,32 @@ def invf(x):
         ],
     )
     def test_tree_only(self, pytree_impl):
-        self.assertEqual(
-            pytree_impl.tree_map_only(int, lambda x: x + 2, [0, "a"]), [2, "a"]
-        )
+        def run_test(typ, pytree):
+            def f(x):
+                return x * 3
+
+            def tree_flatten_only(typ, x):
+                return [x for x in pytree_impl.tree_flatten(x) if isinstance(x, typ)]
+
+            sm1 = sum(map(f, tree_flatten_only(typ, pytree)))
+            sm2 = sum(tuple(tree_flatten_only(typ, pytree_impl.tree_map_only(typ, f, pytree))))
+            self.assertEqual(sm1, sm2)
+
+            def invf(x):
+                return x // 3
+
+            self.assertEqual(
+                pytree_impl.tree_map_only(typ, invf, pytree_impl.tree_map_only(typ, f, pytree)),
+                pytree,
+            )
```

Comment:
assertEqual doesn't seem like a terribly good test here because it doesn't check that you have a torch.Size on both sides, right?